Resync WPT webrtc-encoded-transform

youennf opened the pull request:
#### a3407f99362236534bc033394a192c4f4ed40d9b
<pre>
Resync WPT webrtc-encoded-transform
<a href="https://rdar.apple.com/156895492">rdar://156895492</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=296555">https://bugs.webkit.org/show_bug.cgi?id=296555</a>

Reviewed by Jean-Yves Avenard.

Updated to 6bfecec.

* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-encoded-transform.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-encoded-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-sender-worker-single-frame.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/encoded-frame-clone.https-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/encoded-frame-clone.https.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/encoded-frame-constructors.https-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/encoded-frame-constructors.https.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/encoded-frame-worker.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/idlharness.https.window.js:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/routines.js:
(async createConnectionWithTransform):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-audiolevel-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-audiolevel.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-clone.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-clone.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-metadata.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-metadata.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-receive-cloned.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-receive-cloned.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-send-incoming.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-send-incoming.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-serviceworker-failure.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-serviceworker-failure.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedFrame-timestamps-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedFrame-timestamps-helper.js: Added.
(addHeaderExtensionToSdp):
(async addAbsCaptureTimeAndExchangeOffer):
(async checkAbsCaptureTimeAndExchangeAnswer):
(async exchangeOfferAndListenToOntrack):
(async initiateCall):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedFrame-timestamps.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-clone.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-clone.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-metadata.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-metadata.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-serviceworker-failure.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-serviceworker-failure.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCPeerConnection-sender-worker-single-frame.js: Added.
(onmessage.async event):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/resources/blank.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/resources/serviceworker-failure.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/resources/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/w3c-import.log: Added.

Canonical link: <a href="https://commits.webkit.org/297931@main">https://commits.webkit.org/297931@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a23a17468bc39cd53a2a7f2744dc413e4ca2055d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113488 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33203 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23645 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119669 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/64256 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33815 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41773 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86345 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/41431 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116436 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27010 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102028 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66673 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26277 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20148 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63397 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96393 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20226 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122904 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40506 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30265 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95210 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40897 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98232 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94956 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24224 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40075 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17889 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36670 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40390 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40048 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43361 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41805 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard reviewed (approved):
r=me ; missing some expectations for failing GTK tests and the corresponding bugs created.

youennf committed:
Resync WPT webrtc-encoded-transform
rdar://156895492
https://bugs.webkit.org/show_bug.cgi?id=296555

Reviewed by Jean-Yves Avenard.

Updated to 6bfecec.

* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-encoded-transform.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-encoded-transform.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/RTCRtpScriptTransform-sender-worker-single-frame.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/encoded-frame-clone.https-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/encoded-frame-clone.https.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/encoded-frame-constructors.https-expected.txt: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/encoded-frame-constructors.https.html: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/encoded-frame-worker.js: Removed.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/idlharness.https.window.js:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/routines.js:
(async createConnectionWithTransform):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.https-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/script-transform-sendKeyFrameRequest.https.html:
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-audiolevel-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-audiolevel.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-clone.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-clone.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-metadata.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-metadata.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-receive-cloned.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-receive-cloned.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-send-incoming.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-send-incoming.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-serviceworker-failure.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedAudioFrame-serviceworker-failure.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedFrame-timestamps-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedFrame-timestamps-helper.js: Added.
(addHeaderExtensionToSdp):
(async addAbsCaptureTimeAndExchangeOffer):
(async checkAbsCaptureTimeAndExchangeAnswer):
(async exchangeOfferAndListenToOntrack):
(async initiateCall):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedFrame-timestamps.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-clone.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-clone.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-metadata.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-metadata.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-serviceworker-failure.https-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCEncodedVideoFrame-serviceworker-failure.https.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/RTCPeerConnection-sender-worker-single-frame.js: Added.
(onmessage.async event):
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/resources/blank.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/resources/serviceworker-failure.js: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/resources/w3c-import.log: Added.
* LayoutTests/imported/w3c/web-platform-tests/webrtc-encoded-transform/tentative/w3c-import.log: Added.

Canonical link: https://commits.webkit.org/297931@main
